src/lib.rs

```diff
@@ -33,12 +33,40 @@ assert_eq!(<X as Integer>::to_i32(), 7);
 ```
 
 Documented in each module is the full list of type operators implemented.
-*/
-#![cfg_attr(feature="no_std", no_std)]
-#[cfg(feature="no_std")]
-extern crate core as std;
-
-use std::cmp::Ordering;
+ */
+#![no_std]
+
+use core::cmp::Ordering;
+
+macro_rules! impl_derivable {
+    ($Type: ty) => (
+        impl ::core::cmp::PartialEq for $Type {
+            fn eq(&self, _: &Self) -> bool { match *self {} }
+        }
+        impl ::core::cmp::Eq for $Type { }
+        impl ::core::cmp::PartialOrd for $Type {
+            fn partial_cmp(&self, _: &Self) -> Option<::core::cmp::Ordering> { match *self {} }
+        }
+        impl ::core::cmp::Ord for $Type {
+            fn cmp(&self, _: &Self) -> ::core::cmp::Ordering { match *self {} }
+        }
+        impl ::core::clone::Clone for $Type {
+            fn clone(&self) -> Self { match *self {} }
+        }
+        impl ::core::marker::Copy for $Type {}
+        impl ::core::hash::Hash for $Type {
+            fn hash<H>(&self, _: &mut H) where H: ::core::hash::Hasher { match *self {} }
+        }
+        impl ::core::default::Default for $Type {
+            fn default() -> Self { unreachable!() }
+        }
+        impl ::core::fmt::Debug for $Type {
+            fn fmt(&self, _: &mut ::core::fmt::Formatter) -> ::core::result::Result<(), ::core::fmt::Error> {
+                match *self {}
```

Comment:
Yes, although evidently the match can be optimized better.

https://www.reddit.com/r/rust/comments/43j5fk/what_is_the_meaning_of_match_self/czilais?context=10000